README.md

```diff
@@ -487,91 +487,122 @@ gated on a condition, like tests passing.
+         So that you can push to Gitlab.
+         
+      1) Now that your ssh key is set up, you will need to add Gitlab as a remote. Run
+         ```
+         git remote add gitlab git@ssh.gitlab.dev.explore.data.humancellatlas.org:azul/azul.git
+         ```
+         Run
+         ```
+         git fetch gitlab
+         ```
+         to ensure that your connection is working.
+    
+1) **Prepare for Promotion** (Skip these step if you are deploying without promoting)
+   
+   1) If promoting to `staging` or `prod` you will need to prepare release notes. 
+      Make sure you do this **at least 24 hours in advance**. A link to the release
+      notes document can be found either in the #dcp-ops channel in HCA Slack or in
+      the [release guide document](https://allspark.dev.data.humancellatlas.org/dcp-ops/docs/wikis/SOP:%20Releasing%20new%20Versions%20of%20DCP%20Software)
+      1) To produce the list of changes for the DCP release notes, run 
+         ```
+         git log --pretty=oneline --topo-order --abbrev-commit
+         ```
+         All non-merge commits from the top down to the commit labeled with the
+         most recent `deployed/…` tag represent changes to be deployed and should
+         be mentioned in the release notes.
+      
+      1) You will also need to add the release tag and commit hash which are
+         generated later in this guide.
+      
+   1) From the `azul.stable` clone make sure all of the relevant branches are up to date
+      ```
+      cd azul.stable
+      git checkout SOURCE
+      git pull
+      git checkout TARGET
+      git pull
+      ```
+   
+   1) You should be on the `TARGET` branch. Run
+      ```
+      git merge SOURCE
+      ```
+      and resolve conflicts in necessary. Conflict resolution should only be
+      necessary if cherry-picks occured on the target branch.
+
+    1) The merge may have affected the README. Make sure you're looking at the
+       right version.
+   
+1) **Finishing up Deployment / Promotion**
+   
+   1) Now you need to push to Github. If the build on Gitlab fails, you may need to
+      revert this step.
+      ```
+      git push origin
+      ```
+   
+   1) Finally, push to Gitlab
+      ```
+      git push gitlab
+      ```
+      The build should start immediately. You can monitor its progress from the
+      [Gitlab Pipelines page](https://gitlab.dev.explore.data.humancellatlas.org/azul/azul/pipelines).
+   
+   1) Assuming the build is successful run
+      ```
+      make tag
+      ```
+      and then also run the
+      ```
+      git push ...
+      ```
+      invocation that it echoes.
+      
+      Copy this tag and add it to the release notes (if applicable).
+   
+   1) Invoke the health and version endpoints. Be sure to use the correct 
+      deployment name:
+   
+      ```
+      http https://indexer.TARGET.explore.data.humancellatlas.org/version
```

Comment:
Except `.dev` vs `develop` and `.` vs `prod`. Don't know how you want to handle this.

Reply:
I denormalized so it should be easier to copy / paste. Also skipped prod because of your later comment.

Reply:
I don't understand your reply. 

Anyhow, how do you propose we address the fact that these instructions don't work if `TARGET` is `prod`?